src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -2101,6 +2101,69 @@ internal static BindingRestrictions GetVersionCheck(int expectedVersionNumber)
         }
     }
 
+
+    /// <summary>
+    /// This binder is used to add or remove delegates from event handlers.
+    /// </summary>
+    internal class PSEventDelegateBinder : DynamicMetaObjectBinder
+    {
+        private static readonly PSEventDelegateBinder s_addInstanceBinder = new PSEventDelegateBinder(
+            ExpressionType.Add, @static: false);
+        private static readonly PSEventDelegateBinder s_removeInstanceBinder = new PSEventDelegateBinder(
+            ExpressionType.Subtract, @static: false);
+        private static readonly PSEventDelegateBinder s_addStaticBinder = new PSEventDelegateBinder(
+            ExpressionType.Add, @static: true);
+        private static readonly PSEventDelegateBinder s_removeStaticBinder = new PSEventDelegateBinder(
+            ExpressionType.Subtract, @static: true);
+
+        internal static PSEventDelegateBinder Get(ExpressionType expressionType, bool @static)
+            => expressionType == ExpressionType.Add
+                ? @static ? s_addStaticBinder : s_addInstanceBinder
+                : @static ? s_removeStaticBinder : s_removeInstanceBinder;
+
+        private readonly bool _isAddBinder;
+        private readonly bool _isStatic;
+
+        private PSEventDelegateBinder(ExpressionType expressionType, bool @static)
+        {
+            _isAddBinder = expressionType == ExpressionType.Add;
+            _isStatic = @static;
+        }
+
+        public override DynamicMetaObject Bind(DynamicMetaObject target, DynamicMetaObject[] args)
```

Comment:
Maybe `internal`?

Reply:
I'll check, I think the binders in general have Bind() as a public member?

Reply:
> I'll check, I think the binders in general have Bind() as a public member?

Yeah it's an abstract method.  Can't change accessibility modifier.

Reply:
Yep, `Bind()` is public. Think it has to be, since it's an `override` from the base class that's also public.

Reply:
#Closed.